app/components/dialogs/prioritize-build-modal.js

```diff
@@ -0,0 +1,51 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import eventually from 'travis/utils/eventually';
+import { reads, or, not } from '@ember/object/computed';
+import { task } from 'ember-concurrency';
+import {
+  bindKeyboardShortcuts,
+  unbindKeyboardShortcuts
+} from 'ember-keyboard-shortcuts';
+
+export default Component.extend({
+  keyboardShortcuts: {
+    'esc': 'toggleCloseModal'
+  },
+
+  flashes: service(),
+
+  didInsertElement() {
+    this._super(...arguments);
+    bindKeyboardShortcuts(this);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    unbindKeyboardShortcuts(this);
+  },
+
+  shouldCancelRunningJobs: true,
+
+  item: or('job.build', 'build'),
+
+  increasePriorityTask: task(function* () {
+    yield eventually(this.item, (record) => {
+      record.increasePriority(this.shouldCancelRunningJobs).then(() => {
+        this.flashes.success('The build was successfully prioritized.');
+        this.set('isOpen', false);
+      }, () => {
+        this.flashes.error('An error occurred. The build could not be prioritized.');
+      });
+    });
```

Comment:
@ShiviStar the `task`'s `isRunning` property doesn't reflect the expected value because `eventually` does not return a promise, therefore, the `task` is not even involved in the asynchrony, it just finishes synchronously and doesn't know when the request is done.

Could you please try this code:
```js
    try {
      const record = yield this.item; // this will do the same as `eventually` - resolve the record
      yield record.increasePriority(this.shouldCancelRunningJobs); // this should work for `build`, but won't work for `job` because `job` doesn't have `increasePriority` method
      this.flashes.success('The build was successfully prioritized.');
      this.set('isOpen', false);
    } catch (error) {
      this.flashes.error('An error occurred. The build could not be prioritized.');
    }
```
With this code it should work properly. Please let me know if it doesn't.

Reply:
Hello, @artursmirnov thank you for assisting in this, and yes it is working and resolves both our open issues of removal of eventually and consuming isRunning state. Thank you again.

Reply:
Happy to help! :)